src/ui/Overlay.js

```diff
@@ -3,37 +3,43 @@ import { Footer } from '@pmndrs/branding'
 import { Keys } from './Help'
 import { useStore } from '../store'
 
+import { Loader } from './Loader';
+
 function Ready({ setReady }) {
-  useEffect(() => () => void setReady(true), [])
-  return null
+    useEffect(() => () => void setReady(true), [])
+    return null;
 }
 
 export function Overlay({ children }) {
-  const [ready, setReady] = useState(false)
-  const [clicked, setClicked] = useState(false)
-  const set = useStore((state) => state.set)
+    const [ready, setReady] = useState(false)
+    const [clicked, setClicked] = useState(false)
+    const set = useStore((state) => state.set)
 
-  useEffect(() => {
-    if (clicked && ready) set({ ready: true })
-  }, [ready, clicked])
+    useEffect(() => {
+        if (clicked && ready) set({ ready: true })
+    }, [ready, clicked])
 
-  return (
-    <>
-      <Suspense fallback={<Ready setReady={setReady} />}>{children}</Suspense>
-      <div className={`fullscreen bg ${ready ? 'ready' : 'notready'} ${clicked && 'clicked'}`}>
-        <div className="stack">
-          <Keys style={{ paddingBottom: 20 }} />
-          <a href="#" onClick={() => ready && setClicked(true)}>
-            {!ready ? 'loading' : 'Click to continue'}
-          </a>
-        </div>
-        <Footer
-          date="2. June"
-          year="2021"
-          link1={<a href="https://github.com/pmndrs/react-three-fiber">@react-three/fiber</a>}
-          link2={<a href="https://github.com/pmndrs/racing-game">/racing-game</a>}
-        />
-      </div>
-    </>
-  )
+    return (
+        <>
+            <Suspense fallback={<Ready setReady={setReady} />}>{children}</Suspense>
+            <div className={`fullscreen bg ${ready ? 'ready' : 'notready'} ${clicked && 'clicked'}`}>
+                {!ready ? <Loader /> :
+                    <>
```

Comment:
I was hoping to make use of the `useProgress` hook from https://github.com/pmndrs/drei

and the loader should replace `Ready`
```suggestion
            <Suspense fallback={<Loader setReady={setReady} />}>{children}</Suspense>
            <div className={`fullscreen bg ${ready ? 'ready' : 'notready'} ${clicked && 'clicked'}`}>
            <div className="stack">
```

Reply:
got it, yeah I can do that

Reply:
So I just implemented the loading screen as a fallback so the loading screen now appears after 'Click to continue' is pressed (it will show for maybe a second or two max). Is this what you wanted?

Reply:
Yes that sounds about correct, it's just to make sure everything is loaded by the time the user wants to view the scene.. as a _fallback_